sopel/bot.py

```diff
@@ -573,22 +572,21 @@ def unregister_shutdowns(self, shutdowns):
 
     def register_urls(self, urls):
         for func in urls:
-            for regex in func.url_regex:
-                self.register_url_callback(regex, func)
-                callable_name = getattr(func, "__name__", 'UNKNOWN')
-                LOGGER.debug(
-                    'URL Callback added "%s" for URL pattern "%s"',
-                    callable_name,
-                    regex)
-
-    def unregister_urls(self, urls):
-        if "url_callbacks" in self.memory:
-            for func in urls:
-                regexes = func.url_regex
-                for regex in regexes:
-                    if func == self.memory['url_callbacks'].get(regex):
-                        self.unregister_url_callback(regex, func)
-                        LOGGER.debug('URL Callback unregistered: %r', regex)
+            url_regex = getattr(func, 'url_regex', [])
+            url_lazy_loader = getattr(func, 'url_lazy_loader', None)
+
+            if url_regex:
+                rule = plugin_rules.URLCallback.from_callable(
+                    self.settings, func)
+                self._rules_manager.register_url_callback(rule)
+
+            if url_lazy_loader:
+                try:
+                    rule = plugin_rules.URLCallback.from_callable_lazy(
+                        self.settings, func)
+                    self._rules_manager.register_url_callback(rule)
+                except plugins.exceptions.PluginError as err:
+                    LOGGER.error('Can not register URL callback: %s', err)
```

Comment:
Did I miss this before? I feel like maybe. Whoops.

```suggestion
                    LOGGER.error('Cannot register URL callback: %s', err)
```